From f762b06265dabf9cf7ccc97d2e1c0893f3de63fc Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 4 Jan 2024 09:41:11 +0100
Subject: [PATCH 0867/1141] drm/stm: ltdc: set transparency after plane disable

The plane's opacity should be transparent while the plane
is disabled. This fixes a hardware bug (plane still
displayed even if this plane was disabled).

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: I07510560f6f52a79e55b0395251f07424512a413
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/352880
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index 699e1fb4e4fa..3d79df9b5656 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -1752,6 +1752,9 @@ static void ltdc_plane_atomic_disable(struct drm_plane *plane,
 	/* Disable layer */
 	regmap_write_bits(ldev->regmap, LTDC_L1CR + lofs, LXCR_MASK, 0);
 
+	/* Set the transparency of the layer to the default value */
+	regmap_write_bits(ldev->regmap, LTDC_L1CACR + lofs, LXCACR_CONSTA, 0x00);
+
 	/* Commit shadow registers = update plane at next vblank */
 	if (ldev->caps.plane_reg_shadow)
 		regmap_write_bits(ldev->regmap, LTDC_L1RCR + lofs,
-- 
2.39.2

